### PR TITLE
update containerd for refactor

### LINKED ISSUE #423: Update containerd
There are several code refactoring in containerd recently which may need our side to change as well:
* https://github.com/containerd/containerd/pull/1782
* https://github.com/containerd/containerd/pull/1777
* May be more.

### PR DESCRIPTION
**Pull request by yanxuean**
fix #423
Signed-off-by: yanxuean <yan.xuean@zte.com.cn>

### CODE REVIEW

**Comment by yanxuean** [comment]
The reason all are to pulling pause fail. Please trigger a new test. Thanks. @Random-Liu  
```
=== RUN   TestContainerStats

E1124 07:35:14.263871   24099 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to get sandbox image "gcr.io/google_containers/pause:3.0": failed to pull image "gcr.io/google_containers/pause:3.0": failed to pull image "gcr.io/google_containers/pause:3.0": failed commit on ref "layer-sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c": unexpected commit digest sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, expected sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c: unknown

--- FAIL: TestContainerStats (0.53s)
```

File: vendor.conf [hunk 1]
```diff
@@ -2,7 +2,7 @@ github.com/blang/semver v3.1.0
 github.com/boltdb/bolt e9cf4fae01b5a8ff89d0ec6b32f0d9c9f79aefdd
 github.com/BurntSushi/toml a368813c5e648fee92e5f6c30e3944ff9d5e8895
 github.com/containerd/cgroups f7dd103d3e4e696aa67152f6b4ddd1779a3455a9
-github.com/containerd/containerd v1.0.0-beta.3
+github.com/containerd/containerd 8e09b565a70f534693deaecde136d5d6697df33e
```
Comment (mikebrow) [inline vendor.conf]: When you change containerd's version you also need to change it over here:

https://github.com/kubernetes-incubator/cri-containerd/blob/master/hack/versions#L3

Comment (yanxuean) [reply vendor.conf]: ok

**Comment by yanxuean** [comment]
@mikebrow @Random-Liu   ping for ok-to-test

File: pkg/server/events.go [hunk 2]
```diff
@@ -17,8 +17,9 @@ limitations under the License.
 package server
 
 import (
-	"github.com/containerd/containerd"
-	"github.com/containerd/containerd/api/services/events/v1"
+	containerdio "github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/api/events"
```
[the remark this answers is not in the document]
Comment (yanxuean) [reply pkg/server/events.go]: I will see it

File: vendor.conf (hunk 1 above)
Comment (yanxuean) [reply vendor.conf]: done

**Comment by yanxuean** [comment]
I don't pull pause image success. Seem like the containerd have some problem.
When I pull pause image with ctr, I is fail yet. 
@mikebrow @Random-Liu  Are there thoughts?

```
root@ubuntu:cri-containerd# crictl pull gcr.io/google_containers/pause:3.0
ERRO[1967] (*service).Write failed                       error="unexpected commit digest sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, expected sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c" expected=sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c ref="layer-sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c"
ERRO[1967] (*service).Write failed                       error="rpc error: code = Canceled desc = context canceled" expected=sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4 ref="layer-sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
E1127 14:47:49.293271   14519 instrumented_service.go:239] PullImage "gcr.io/google_containers/pause:3.0" failed, error: failed to pull image "gcr.io/google_containers/pause:3.0": failed commit on ref "layer-sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c": unexpected commit digest sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, expected sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c: unknown
FATA[0002] pulling image failed: rpc error: code = Unknown desc = failed to pull image "gcr.io/google_containers/pause:3.0": failed commit on ref "layer-sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c": unexpected commit digest sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, expected sha256:f112334343777b75be77ec1f835e3bbbe7d7bd46e27b6a2ae35c6b3cfea0987c: unknown 
```

```
root@ubuntu:docker# ctr images pull gcr.io/google_containers/pause:3.0
gcr.io/google_containers/pause:3.0: resolving      |--------------------------------------| 
elapsed: 1.5 s                      total:   0.0 B (0.0 B/s)                                         
ctr: failed commit on ref "layer-sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4": unexpected commit digest sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, expected sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4: unknown
```

File: pkg/server/events.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yanxuean) [reply pkg/server/events.go]: I saw the modification of events.  It is correct.:)

**Comment by yanxuean** [comment]
Really a sad news, I fell into the trap of #1791.:) 
Fortunately, we finally solved it. 
Thanks @mikebrow @Random-Liu 

I have a question. When can  we sent a request of kubernetes-incubator member ?

**Comment by Random-Liu** [comment]
https://github.com/containerd/containerd/pull/1792/files@yanxuean We should update to latest containerd version to include the fix

**Comment by Random-Liu** [comment]
> I have a question. When can we sent a request of kubernetes-incubator member ?

I can do that for you! And do you also want to be a member? @miaoyq

**Comment by Random-Liu** [comment]
Close this one, since I've picked up the commits in this PR.
@yanxuean

**Comment by miaoyq** [comment]
@Random-Liu yeah! I'm glad to be a member, it will make me do more for kubernetes-incubator😊

**Comment by yanxuean** [comment]
>I can do that for you! And do you also want to be a member? @miaoyq

@Random-Liu  Thank you. We also are member of kubernetes incubator

**Pull request closed without merging**